pkg/memory/queue.go

```diff
@@ -0,0 +1,48 @@
+package memory
+
+import (
+	"fmt"
+	"scratchdata/pkg/queue"
+	"sync"
+)
+
+var (
+	_ queue.QueueBackend = (*Queue)(nil)
```

Comment:
What does this do?

Reply:
It makes compilation fail if the `queue.QueueBackend` interface changes and `Queue` doesn't implement the changes.

Reply:
Okay. When we actually start using these objects in `main.go` this will no longer be needed. (We can delete it then.)